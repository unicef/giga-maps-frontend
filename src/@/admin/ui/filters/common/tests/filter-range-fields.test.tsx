import { describe, expect, test, } from '@jest/globals';
import { fireEvent, render, screen } from '@testing-library/react';
import FilterRangeFields from '../filter-range-fields';
import FilterCommonFields from '../filter-common-fields';
import { useStore } from 'effector-react';
import { $filterColumnList, $filterTypeChoices, $filterQueryParamsChoices, $formFilterData } from '~/@/admin/models/filter-list.model';
import { testWrapper } from '~/tests/jest-wrapper';


jest.mock('effector-react', () => ({
  useStore: jest.fn()
}));

describe('FilterRangeFields', () => {

  beforeEach(() => {
    (useStore as jest.Mock).mockImplementation((store) => {
      if (store === $filterColumnList) {
        return [];
      }
      if (store === $filterTypeChoices) {
        return {};
      }
      if (store === $filterQueryParamsChoices) {
        return {};
      }
      if (store === $formFilterData) {
        return {};
      }
    });
  });
  test('renders FilterRangeFields and take a snapshop', () => {
    const { asFragment } = render(<FilterRangeFields />);
    expect(asFragment()).toMatchSnapshot();
  });


  test('renders FilterCommonFields and take a snapshop, edit mode true', () => {
    const { asFragment } = render(<FilterCommonFields isEditMode={true} />)
    expect(asFragment()).toMatchSnapshot();
  })

})



describe('FilterCommonFields', () => {
  beforeEach(() => {
    (useStore as jest.Mock).mockImplementation((store) => {
      if (store === $filterColumnList) {
        return [{
          id: 1,
          name: 'test',
          label: 'Test Column',
          options: {
            applicable_filter_types: {
              range: ['param1', 'param2']
            }
          }
        }];
      }
      if (store === $filterTypeChoices) {
        return {
          typeChoicesList: [
            { value: 'range', label: 'Range' },
            { value: 'select', label: 'Select' }
          ]
        };
      }
      if (store === $filterQueryParamsChoices) {
        return {
          queryParamsList: [
            { value: 'param1', label: 'Parameter 1' },
            { value: 'param2', label: 'Parameter 2' }
          ]
        };
      }
      if (store === $formFilterData) {
        return {
          code: 'test-code',
          column_configuration: '1',
          type: 'range',
          name: 'Test Filter',
          description: 'Test Description',
          query_param_filter: 'param1'
        };
      }
    });
  });

  test('renders all form fields', () => {
    const { getByText } = render(testWrapper(
      <FilterCommonFields isEditMode={false} />
    ));

    expect(getByText('Unique Code')).toBeInTheDocument();
    expect(getByText('Parameter')).toBeInTheDocument();
    expect(getByText('Filter type')).toBeInTheDocument();
    expect(getByText('Filter name')).toBeInTheDocument();
    expect(getByText('Filter description')).toBeInTheDocument();
  });

  test('handles input changes', async () => {
    const { getByPlaceholderText } = render(testWrapper(
      <FilterCommonFields isEditMode={false} />
    ));

    const nameInput = getByPlaceholderText('Enter filter name');
    fireEvent.change(nameInput, { target: { value: 'Test Filter' } });
    expect(nameInput.value).toBe('Test Filter');
  });
});
