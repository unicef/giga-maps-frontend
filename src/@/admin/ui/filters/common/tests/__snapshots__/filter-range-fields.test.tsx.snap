// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRangeFields renders FilterCommonFields and take a snapshop, edit mode true 1`] = `
<DocumentFragment>
  .c0 {
  color: #c0c0c0;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
}

.c0.center-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance::before {
  background-color: #c0c0c0;
}

.c0 .cds--checkbox-label::before,
.c0 .cds--radio-button__appearance,
.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance {
  border-color: #c0c0c0;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon path:not([data-icon-path]):not([fill=none]),
.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon {
  fill: #c0c0c0;
}

<div
    class="c0"
  >
    Unique Code
  </div>
  .c0 .cds--text-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--text-input::placeholder {
  color: #f5f5f5;
}

.c0 .cds--text-input:disabled {
  color: #f5f5f5;
  -webkit-text-fill-color: #f5f5f5;
  opacity: 0.4;
}

.c0 input:-webkit-autofill,
.c0 input:-webkit-autofill:hover,
.c0 input:-webkit-autofill:focus,
.c0 textarea:-webkit-autofill,
.c0 textarea:-webkit-autofill:hover,
.c0 textarea:-webkit-autofill:focus {
  box-shadow: 0 0 0 1000px #333333 inset;
  -webkit-text-fill-color: #f5f5f5;
}

<div
    class="cds--form-item c0 cds--text-input-wrapper"
  >
    <div
      class="cds--text-input__label-wrapper"
    />
    <div
      class="cds--text-input__field-outer-wrapper"
    >
      <div
        class="cds--text-input__field-wrapper"
      >
        <input
          class="cds--text-input"
          disabled=""
          id="layer-code"
          name="code"
          placeholder="Enter layer unique code"
          required=""
          title="Enter layer unique code"
          type="text"
        />
        <span
          class="cds--text-input__counter-alert"
          role="alert"
        />
      </div>
    </div>
  </div>
  .c0 {
  margin-top: 0.5rem;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--select-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--list-box__label {
  color: #f5f5f5;
}

.c0 svg {
  fill: #f5f5f5;
}

<div
    class="cds--form-item c0"
  >
    <div
      class="cds--select"
    >
      <label
        class="cds--label"
        for="parameter-select"
      >
        Parameter
      </label>
      <div
        class="cds--select-input__wrapper"
      >
        <select
          class="cds--select-input"
          id="parameter-select"
          name="column_configuration"
          placeholder="Choose parameter type"
          required=""
          title=""
        >
          <option
            class="cds--select-option"
            value=""
          >
            Choose parameter
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="cds--select__arrow"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
          />
        </svg>
      </div>
    </div>
  </div>
  .c0 {
  margin-top: 0.5rem;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--select-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--list-box__label {
  color: #f5f5f5;
}

.c0 svg {
  fill: #f5f5f5;
}

<div
    class="cds--form-item c0"
  >
    <div
      class="cds--select"
    >
      <label
        class="cds--label"
        for="filter-type-select"
      >
        Filter type
      </label>
      <div
        class="cds--select-input__wrapper"
      >
        <select
          class="cds--select-input"
          id="filter-type-select"
          name="type"
          placeholder="Choose filter type"
          required=""
          title=""
        >
          <option
            class="cds--select-option"
            value=""
          >
            Choose filter type
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="cds--select__arrow"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
          />
        </svg>
      </div>
    </div>
  </div>
  .c0 {
  color: #c0c0c0;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
}

.c0.center-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance::before {
  background-color: #c0c0c0;
}

.c0 .cds--checkbox-label::before,
.c0 .cds--radio-button__appearance,
.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance {
  border-color: #c0c0c0;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon path:not([data-icon-path]):not([fill=none]),
.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon {
  fill: #c0c0c0;
}

<div
    class="c0"
  >
    Filter name
  </div>
  .c0 .cds--text-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--text-input::placeholder {
  color: #f5f5f5;
}

.c0 .cds--text-input:disabled {
  color: #f5f5f5;
  -webkit-text-fill-color: #f5f5f5;
  opacity: 0.4;
}

.c0 input:-webkit-autofill,
.c0 input:-webkit-autofill:hover,
.c0 input:-webkit-autofill:focus,
.c0 textarea:-webkit-autofill,
.c0 textarea:-webkit-autofill:hover,
.c0 textarea:-webkit-autofill:focus {
  box-shadow: 0 0 0 1000px #333333 inset;
  -webkit-text-fill-color: #f5f5f5;
}

<div
    class="cds--form-item c0 cds--text-input-wrapper"
  >
    <div
      class="cds--text-input__label-wrapper"
    />
    <div
      class="cds--text-input__field-outer-wrapper"
    >
      <div
        class="cds--text-input__field-wrapper"
      >
        <input
          class="cds--text-input"
          id="filter-name"
          name="name"
          placeholder="Enter filter name"
          required=""
          title="Enter filter name"
          type="text"
        />
        <span
          class="cds--text-input__counter-alert"
          role="alert"
        />
      </div>
    </div>
  </div>
  .c0 {
  color: #c0c0c0;
  margin-top: 1rem;
  margin-bottom: 0.5rem;
  font-size: 0.75rem;
}

.c0.center-items {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance::before {
  background-color: #c0c0c0;
}

.c0 .cds--checkbox-label::before,
.c0 .cds--radio-button__appearance,
.c0 .cds--radio-button:checked+.cds--radio-button__label .cds--radio-button__appearance {
  border-color: #c0c0c0;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon path:not([data-icon-path]):not([fill=none]),
.c0 .cds--btn--ghost.cds--btn--icon-only .cds--btn__icon {
  fill: #c0c0c0;
}

<div
    class="c0"
  >
    Filter description
  </div>
  .c0 .cds--text-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--text-input::placeholder {
  color: #f5f5f5;
}

.c0 .cds--text-input:disabled {
  color: #f5f5f5;
  -webkit-text-fill-color: #f5f5f5;
  opacity: 0.4;
}

.c0 input:-webkit-autofill,
.c0 input:-webkit-autofill:hover,
.c0 input:-webkit-autofill:focus,
.c0 textarea:-webkit-autofill,
.c0 textarea:-webkit-autofill:hover,
.c0 textarea:-webkit-autofill:focus {
  box-shadow: 0 0 0 1000px #333333 inset;
  -webkit-text-fill-color: #f5f5f5;
}

<div
    class="cds--form-item c0 cds--text-input-wrapper"
  >
    <div
      class="cds--text-input__label-wrapper"
    />
    <div
      class="cds--text-input__field-outer-wrapper"
    >
      <div
        class="cds--text-input__field-wrapper"
      >
        <input
          class="cds--text-input"
          id="filter-description"
          name="description"
          placeholder="Enter description"
          title="Enter description"
          type="text"
        />
        <span
          class="cds--text-input__counter-alert"
          role="alert"
        />
      </div>
    </div>
  </div>
  .c0 {
  margin-top: 0.5rem;
}

.c0 .cds--label {
  color: #c0c0c0;
}

.c0 .cds--select-input {
  background: #333333;
  color: #f5f5f5;
}

.c0 .cds--list-box__label {
  color: #f5f5f5;
}

.c0 svg {
  fill: #f5f5f5;
}

<div
    class="cds--form-item c0"
  >
    <div
      class="cds--select"
    >
      <label
        class="cds--label"
        for="filter-type-select"
      >
        Query Param filter
      </label>
      <div
        class="cds--select-input__wrapper"
      >
        <select
          class="cds--select-input"
          id="filter-type-select"
          name="query_param_filter"
          placeholder="Choose query param filter"
          required=""
          title=""
        >
          <option
            class="cds--select-option"
            value=""
          >
            Choose query param
          </option>
        </select>
        <svg
          aria-hidden="true"
          class="cds--select__arrow"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 16 16"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FilterRangeFields renders FilterRangeFields and take a snapshop 1`] = `<DocumentFragment />`;
